### PR TITLE
Feat/delete UI

### PR DESCRIPTION
**Pull request by zenovore**
**What this PR does / why we need it**: Introduce the UI of deletion of model, model version and prediction jobs, with the API on #389 

**Which issue(s) this PR fixes**: 

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
1. Add Delete Button on model list page
2. Add Delete Button on model version list page
3. Add Terminate/Stop Button on prediction job list page
4. Popup dialog component for each deletion feature showing entity that blocking the deletion process, or the 
related entity that will get deleted from the model/model version

```

**Checklist**

- [ ] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [x] Updated documentation
- [x] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: ui/src/components/modals/DeleteModelModal.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      callback();
+    }
+  }, [isLoaded, closeModal, callback]);
+
+  useEffect(() => {
+    const activeItem = [];
+    // const inactiveItem = [];
+    if (versions.isLoaded){
+      for (const item of versions.data){
+        const activeEnd = item.endpoints.filter(
```
Comment (krithika369) [inline ui/src/components/modals/DeleteModelModal.js:56]: Nit: `activeEnd` -> `activeVersionEndpoints` is clearer.

File: ui/src/components/modals/DeleteModelModal.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import { EuiConfirmModal, EuiFieldText, EuiOverlayMask, EuiProgress } from "@elastic/eui";
+import { useMerlinApi } from "../../hooks/useMerlinApi";
+import mocks from "../../mocks";
+
+const DeleteModelModal = ({
+  model,
+  callback,
+  closeModal
+}) => {
+  const [activeEndpoint, setActiveEndpoint] = useState([])
```
Comment (krithika369) [inline ui/src/components/modals/DeleteModelModal.js:27]: Nit: Perhaps `activeEndpoints` and `setActiveEndpoints` (in plural) are more accurate.

File: ui/src/components/modals/DeleteModelModal.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            <div>
+            {activeEndpoint.length > 0 ? (
+              <span>
+                You cannot delete this Model because there are <b> {activeEndpoint.length} Endpoints</b> currently using the model version associated with it. 
+                <br/> <br/> If you still wish to delete this model, please <b>Undeploy</b> Endpoints that use this version. <br/>
+              </span>
+            ) : (
+              <div>
+                  You are about to delete model <b>{model.name}</b>. This action cannot be undone. 
+                  <br/> <br/> Please note that all the related entity of this model (<b>Model version</b> and <b>Model Version Endpoint)</b> will be <b>deleted</b>.
```
Comment (krithika369) [inline ui/src/components/modals/DeleteModelModal.js:111]: Nit: all the related `entity` -> all the related `entities` (plural is more accurate).

File: ui/src/components/modals/DeleteModelPyFuncV2Modal.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import { EuiConfirmModal, EuiFieldText, EuiOverlayMask, EuiProgress } from "@elastic/eui";
+import { useMerlinApi } from "../../hooks/useMerlinApi";
+import mocks from "../../mocks";
+
+const DeleteModelPyFuncV2Modal = ({
+  model,
+  callback,
+  closeModal
+}) => {
+  const [activeJob, setActiveJob] = useState([])
```
Comment (krithika369) [inline ui/src/components/modals/DeleteModelPyFuncV2Modal.js:27]: Nit: Same comment as other files - these names too, can be in plural as they are dealing with lists (`[]`).

File: ui/src/components/modals/DeleteModelPyFuncV2Modal.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+                    placeholder={model.name}
+                    value={deleteConfirmation}
+                    onChange={(e) => setDeleteConfirmation(e.target.value)}
+                    isInvalid={deleteConfirmation !== model.name} />  
+              </div>
+            )}
+            {activeJob.length === 0 && inactiveJob.length > 0 && (
+                <span>Deleting this Model will also delete {inactiveJob.length} <b>Inactive</b> Prediction Jobs using this model. <br/> <br/></span>
+            )}
+            {/* <JobsTableModelVersion jobs={activeJob.length > 0 ? activeJob : inactiveJob} isLoaded={jobs.isLoaded} error={jobs.error} /> */}
```
Comment (krithika369) [inline ui/src/components/modals/DeleteModelPyFuncV2Modal.js:107]: Is the decision to NOT show the individual job info for delete models (as there can be too many)? If so, I agree with this. Let's remove the commented out lines though.

File: ui/src/components/modals/DeleteModelVersionModal.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                    isInvalid={deleteConfirmation !== `${model.name}-version-${version.id}`} />  
+              </div>
+            )}
+            <br></br>
+            {activeEndpoint.length === 0 && inactiveEndpoint.length > 0 && (
+                <p>Deleting this Model Version will also delete {inactiveEndpoint.length} <b>Failed</b> Endpoints using this version. </p>
+            )}
+
+            <EndpointsTableModelVersion
+              endpoints={activeEndpoint.length > 0 ? activeEndpoint : inactiveEndpoint}
```
Comment (krithika369) [inline ui/src/components/modals/DeleteModelVersionModal.js:109]: Can we move this `<EndpointsTableModelVersion />`  under the previous check for `{activeEndpoint.length > 0 ? (`, along with the modal's message? Having this here is a little bit misleading (makes it look like we will give preference to displaying active endpoints, but actually, if there are active endpoints, the message/action items on the user is also different).

File: ui/src/components/modals/DeleteModelVersionPyFuncV2Modal.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+                    value={deleteConfirmation}
+                    onChange={(e) => setDeleteConfirmation(e.target.value)}
+                    isInvalid={deleteConfirmation !== `${model.name}-version-${version.id}`} />  
+              </div>
+            )}
+            <br></br>
+            {activeJob.length === 0 && inactiveJob.length > 0 && (
+                <span>Deleting this Model Version will also delete {inactiveJob.length} <b>Inactive</b> Prediction Jobs using this version. <br/> <br/></span>
+            )}
+            <JobsTableModelVersion jobs={activeJob.length > 0 ? activeJob : inactiveJob} isLoaded={jobs.isLoaded} error={jobs.error} />
```
Comment (krithika369) [inline ui/src/components/modals/DeleteModelVersionPyFuncV2Modal.js:110]: Same comment as `ui/src/components/modals/DeleteModelVersionModal.js` on the placement of this `<JobsTableModelVersion />` block.

File: ui/src/job/JobListTable.js [hunk 8]
```diff
@@ -206,20 +210,19 @@ const JobListTable = ({ projectId, modelId, jobs, isLoaded, error }) => {
...
-          {(item.status === "pending" || item.status === "running") && (
             <EuiFlexItem grow={false} key={`stop-job-${item.id}`}>
               <EuiButtonEmpty
                 onClick={() => {
                   setCurrentJob(item);
                   toggleStopPredictionJobModal(true);
                 }}
                 color="danger"
-                iconType="minusInCircle"
+                iconType={isActiveJob(item.status) ? 'minusInCircle' : 'trash'}
```
Comment (krithika369) [inline ui/src/job/JobListTable.js:220]: Nit: Can we stick to double quotes everywhere?

File: ui/src/pages/version/components/modal/EndpointsTableModelVersion.js [hunk 9]
```diff
@@ -0,0 +1,132 @@
+/**
```
Comment (krithika369) [inline ui/src/pages/version/components/modal/EndpointsTableModelVersion.js:1]: Nit on the naming of these 2 files:
* EndpointsTableModelVersion.js -> ModelVersionEndpointsTable
* JobsTableModelVersion.js -> ModelVersionJobsTable

I feel that starting the file name with `ModelVersion...` is more descriptive of the thing that comes after (`JobsTable` / `EndpointsTable`).

**Review by krithika369 — COMMENTED** [review]
@zenovore, thanks for this PR. It largely looks good to me - great work! Left some comments.

Will let @deadlycoconuts / @ariefrahmansyah have another look.

**Comment by deadlycoconuts** [comment]
Thanks a lot for this PR! I don't think I have any other comments on this; let's try to get the API changes merged first before rebasing this one on top of it! :)

File: ui/src/pages/version/components/modal/ModelVersionEndpointsTable.js [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+        return "orange";
+      case "pending":
+        return "gray";
+      case "terminating":
+        return "warning";
+      case "terminated":
+        return "warning";
+      case "failed":
+        return "danger";
+      case "failed_submission":
```
Comment (krithika369) [inline ui/src/pages/version/components/modal/ModelVersionEndpointsTable.js:46]: We don't have all these states for the endpoint status, right? It's just these:

https://github.com/caraml-dev/merlin/blob/main/db-migrations/1_schema_creation.up.sql#L15

**Review by krithika369 — APPROVED** [review]
Left one new comment. Besides that, LGTM! Thanks, @zenovore ! Please feel free to merge this PR after the API one is merged, and after reviewing the pending comment.

**Review by deadlycoconuts — APPROVED** [review]
Thanks a lot for adding the docs! I just left a couple of tiny comments but otherwise this PR looks good to go! 🚀🚀🚀

File: docs/user-guide/model_version_deletion.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
+# Model Version Deletion
+
+A Merlin model version can be deleted only if it is not serving any endpoints, does not have any deployed endpoints, or if the model has the `pyfunc_v2` type, the model version must not have any active prediction jobs. Deleting a model version will result in purging the model version and its related entities, such as endpoints or prediction jobs, from the Merlin database. This action is **irreversible**.
+
+If there are any active prediction jobs or endpoints on the related model, the deletion will be prohibited.
```
[the remark this answers is not in the document]
Comment (zenovore) [reply docs/user-guide/model_version_deletion.md:5]: I'll change this to 
`Model version with related active prediction jobs or endpoints can not be deleted.` for model version and `A model with associated model versions that have active prediction jobs or endpoints cannot be deleted.
` for model

**Pull request merged**